specutils/tests/test_loaders.py

```diff
@@ -662,27 +662,44 @@ def test_tabular_fits_maskheader(tmp_path):
     disp = np.linspace(1, 1.2, 21) * u.AA
     flux = np.random.normal(0., 1.0e-14, disp.shape[0]) * u.Jy
     hdr = fits.header.Header({'TELESCOP': 'Leviathan', 'APERTURE': 1.8,
-                              'OBSERVER': 'Parsons', 'NAXIS': 1, 'NAXIS1': 8})
+                              'OBSERVER': 'Parsons'})
 
     spectrum = Spectrum1D(flux=flux, spectral_axis=disp, meta={'header': hdr})
     tmpfile = str(tmp_path / '_tst.fits')
     spectrum.write(tmpfile, format='tabular-fits')
 
     # Read it in and check against the original
     with fits.open(tmpfile) as hdulist:
+
+        # Test HDU0 header
         assert hdulist[0].header['NAXIS'] == 0
+        assert hdulist[0].header['OBSERVER'] == 'Parsons'
+
+        # keys relevant to datashape are in HDU1 header
         assert hdulist[1].header['NAXIS'] == 2
         assert hdulist[1].header['NAXIS2'] == disp.shape[0]
+
+        # Test legacy behaviour of metadata is HDU1 header
```

Comment:
Not sure I understand "metadata is HDU1 header" here – "metadata written to HDU1 header" perhaps?